### ISSUE TITLE
Should reduction match on amount?

### ISSUE THREAD

**Issue by blais** [beancount/beancount#338]
**[Original report](https://bitbucket.org/blais/beancount/issue/338) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

bean-check complains about the following but as a human it looks pretty obviou to me what I mean.  Should beancount take the amount into consideration?

```
Ambiguous matches for "43.00 CAD {0.8891 USD}": -24.74 CAD {0.8891 USD, 2014-10-26}, -43.00 CAD {0.8891 USD, 2014-10-27}
Ambiguous matches for "8.00 EUR {1.1358 USD}": -8.00 EUR {1.1358 USD, 2015-09-18}, -218.00 EUR {1.1358 USD, 2015-09-19}
```

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

That's also an interesting extension idea.

**Comment by blais** [comment]
TL;DR: If the units number matches, use that to further disambiguate the matching.
Make this feature optional (add option).
